src/libraries/System.Net.Sockets/src/System/Net/Sockets/SafeSocketHandle.Unix.cs

```diff
@@ -116,11 +116,17 @@ private void SetHandleNonBlocking()
             // We don't care about synchronization because this is idempotent
             if (!_underlyingHandleNonBlocking)
             {
-                AsyncContext.SetNonBlocking();
-                _underlyingHandleNonBlocking = true;
+                // Calls back into MarkNonBlocking()
+                AsyncContext.SetNonBlockingIfNotSet();
```

Comment:
I think we can remove `_underlyingHandleNonBlocking` and use `AsyncContext.SetNonBlockingIfNotSet` to set, and `AsyncContext?.IsNonBlocking` to get?